src/realtime/realtime-note/realtime-note.service.ts

```diff
@@ -38,20 +67,31 @@ export class RealtimeNoteService {
   private async createNewRealtimeNote(note: Note): Promise<RealtimeNote> {
     const initialContent = (await this.revisionsService.getLatestRevision(note))
       .content;
-    const realtimeNote = new RealtimeNote(note.id, initialContent);
-    realtimeNote.on('destroy', () => {
-      this.noteIdToRealtimeNote.delete(note.id);
+    const realtimeNote = this.realtimeNoteRepository.create(
+      note,
+      initialContent,
+    );
+    realtimeNote.on('beforeDestroy', () => {
+      this.saveRealtimeNote(realtimeNote);
     });
-    this.noteIdToRealtimeNote.set(note.id, realtimeNote);
+    this.startPersistTimer(realtimeNote);
     return realtimeNote;
   }
 
   /**
-   * Retrieves a {@link RealtimeNote} that is linked to the given {@link Note} id.
-   * @param noteId The id of the {@link Note}
-   * @return A {@link RealtimeNote} or {@code undefined} if no instance is existing.
+   * Starts a {@link PeriodicTimer timer} that persists the realtime note in a periodic interval depending on the {@link AppConfig}.
+
+   * @param realtimeNote The realtime note for which the timer should be started
    */
-  public getRealtimeNote(noteId: string): RealtimeNote | undefined {
-    return this.noteIdToRealtimeNote.get(noteId);
+  private startPersistTimer(realtimeNote: RealtimeNote): void {
+    Optional.of(this.appConfig.persistInterval)
+      .filter((value) => value > 0)
+      .ifPresent((persistInterval) => {
+        const timer = new PeriodicTimer(
```

Comment:
Can't we use [`schedulerRegistry.addInterval`](https://docs.nestjs.com/techniques/task-scheduling#dynamic-intervals) instead of rolling our own class?

Reply:
Didn't know it existed :shrug: